### PR TITLE
IOS background sync - fix

### PR DESCRIPTION
**Pull request by juanky201271**
If the background task is fired but the App is closed, we need to open/load the wallet file in order to sync in background.

Can you review this, please... @LukasKorba Thanks for your help.

### CODE REVIEW

**Review by LukasKorba — APPROVED** [review]

**Comment by juanky201271** [comment]
Thanks, @LukasKorba

**Review by LukasKorba — APPROVED** [review]

**Review by LukasKorba — APPROVED** [review]

**Review by LukasKorba — APPROVED** [review]

File: ios/ZingoMobile/AppDelegate.m [hunk 1]
```diff
@@ -235,11 +295,22 @@ - (void)handleBackgroundTask {
...
+      // the device have Wifi.
       isConnectedToWifi = true;
     } else {
       isConnectedToWifi = false;
     }
-    NSLog(@"BGTask isConnectedToWifi %@", isConnectedToWifi ? @"true" : @"false");
+
+    UIDeviceBatteryState currentState = [[UIDevice currentDevice] batteryState];
+
+    if (currentState == UIDeviceBatteryStateCharging) {
```
Comment (juanky201271) [inline ios/ZingoMobile/AppDelegate.m:306]: @LukasKorba do you think this is a good idea? We have two restrictions for BS, and we were checking only one: `Wifi`. I thought could be a good idea to check the other: `Charging`.

File: ios/ZingoMobile/AppDelegate.m [hunk 2]
```diff
@@ -275,24 +346,37 @@ - (void)registerTasks {
...
-    NSLog(@"BEFORE RUN TASKS - Time Remaining: %f", [[UIApplication sharedApplication] backgroundTimeRemaining]);
-    if ([[UIApplication sharedApplication] backgroundTimeRemaining] > 1000000000) {
-        NSLog(@"BGTask startBackgroundTask: time remainig TOO cracy high %f", [[UIApplication sharedApplication] backgroundTimeRemaining]);
+    if (!isCharging) {
+        // save info in background json
+        NSTimeInterval timeStamp = [[NSDate date] timeIntervalSince1970];
+        // NSTimeInterval is defined as double
+        NSNumber *timeStampObj = [NSNumber numberWithDouble: timeStamp];
+        NSString *timeStampStr = [timeStampObj stringValue];
+        NSString *jsonBackgroud = [NSString stringWithFormat: @"%@%@%@%@%@%@%@", @"{\"batches\": \"", @"0", @"\", \"message\": \"", @"No plug-in KO.", @"\", \"date\": \"", timeStampStr, @"\"}"];
```
Comment (juanky201271) [inline ios/ZingoMobile/AppDelegate.m:376]: @LukasKorba and here I stored a simple message for the user/tester, to know why last night the BS didn't fire.... please tell me if I'm doing something wrong with this.

File: app/translations/en.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "zingo": "Zingo!",
-  "version": "zingo-1.3.3 (131)",
+  "version": "zingo-1.3.4 (137)",
```
Comment (zancas) [inline app/translations/en.json:3]: Is 137 the right number?

**Review by zancas — APPROVED** [review]
Rubber Stamp

File: app/translations/en.json (hunk 3 above)
Comment (juanky201271) [reply app/translations/en.json:3]: yeap, you can see it in your Testflight info in your iPhone.

**Pull request merged**